.github/workflows/runtime_build_and_test.yml

```diff
@@ -301,10 +301,12 @@ jobs:
           path: |
             **/node_modules
           key: runtime-and-compiler-node_modules-v6-${{ runner.arch }}-${{ runner.os }}-${{ hashFiles('yarn.lock', 'compiler/yarn.lock') }}
-          restore-keys: |
-            runtime-and-compiler-node_modules-v6-${{ runner.arch }}-${{ runner.os }}-
-            runtime-and-compiler-node_modules-v6-
```

Comment:
This leads to growing the cache indefinitely when packages changes: https://github.com/mui/material-ui/pull/29040

Since we don't actively monitor install times, it's safer to deopt in the rare case instead of permanently deopting after dependencies have changed multiple times.